_source/_posts/2021-12-03-angular-dynamic-components.md

```diff
@@ -0,0 +1,773 @@
+---
+layout: blog_post
+title: "Loading Components Dynamically in an Angular App"
+author: alisa-duncan
+by: advocate
+communities: [javascript]
+description: "A tutorial that shows how to create an Angular app with dynamic component loading."
+tags: [javascript, typescript, angular]
+tweets:
+- "Use dynamic component loading for ultimate flexibility! Build an @angular app with dynamic components."
+- "Showing components conditionally is a breeze with dynamic components in @angular. See how in this tutorial."
+image: blog/angular-dynamic-components/social.jpg
+type: awareness
+---
+
+Businesses have unique and complex needs. In addition to the user or organization-specific data to show, there might be a need to display different views and content conditionally. The conditions might include the user's role or which department they belong to.  The information about a user might be part of the authenticated user's [ID token as a profile claim](/blog/2017/07/25/oidc-primer-part-1).
+
+In [Angular](https://angular.io), you can show different components or even parts of templates conditionally using built-in directives such as `*ngIf`. But as the number of conditions to evaluate or the differences in the view changes, managing the correct view to display becomes difficult. These types of scenarios are where [dynamic components](https://angular.io/guide/dynamic-component-loader) are helpful. Angular has the mechanics to load components at runtime so you can dynamically display content. 
```

Comment:
This line is bit confusing to read IMO. May be breaking it further down will help

Reply:
@deepu105 Can you help me understand if there's a particular sentence that's confusing or if it's this whole paragraph? You might have had a better LOC attributed to the suggestion, but I think I lost it when I committed other changes. Sorry about that.

Reply:
Well it's not a big deal, may be its just me. I thought "But as the number of conditions to evaluate or the differences in the view changes, managing the correct view to display becomes difficult." could be clearer but please ignore me if you think it's alright